exercises/all-your-bases/tests/all-your-bases.rs

```diff
@@ -0,0 +1,213 @@
+}
+
+#[test]
+#[ignore]
+fn trinary_to_hexadecimal() {
+    let input_base = 3;
+    let input_digits = &[1, 1, 2, 0];
+    let output_base = 16;
+    let output_digits = &[2, 10];
+    assert_eq!(ayb::convert(input_digits, input_base, output_base).unwrap().as_slice(),
+               output_digits);
+}
+
+#[test]
+#[ignore]
+fn hexadecimal_to_trinary() {
+    let input_base = 16;
+    let input_digits = &[2, 10];
+    let output_base = 3;
+    let output_digits = &[1, 1, 2, 0];
+    assert_eq!(ayb::convert(input_digits, input_base, output_base).unwrap().as_slice(),
+               output_digits);
+}
+
+#[test]
+#[ignore]
+fn fifteen_bit_integer() {
+    let input_base = 97;
+    let input_digits = &[3, 46, 60];
+    let output_base = 73;
+    let output_digits = &[6, 10, 45];
+    assert_eq!(ayb::convert(input_digits, input_base, output_base).unwrap().as_slice(),
+               output_digits);
+}
+
+#[test]
+#[ignore]
+fn empty_list() {
+    let input_base = 2;
+    let input_digits = &[];
+    let output_base = 10;
+    let output_digits = &[];
+    assert_eq!(ayb::convert(input_digits, input_base, output_base).unwrap().as_slice(),
+               output_digits);
+}
+
+#[test]
+#[ignore]
+fn single_zero() {
+    let input_base = 10;
+    let input_digits = &[0];
+    let output_base = 2;
+    let output_digits = &[];
+    assert_eq!(ayb::convert(input_digits, input_base, output_base).unwrap().as_slice(),
+               output_digits);
+}
+
+#[test]
+#[ignore]
+fn multiple_zeros() {
+    let input_base = 10;
+    let input_digits = &[0, 0, 0];
+    let output_base = 2;
+    let output_digits = &[];
+    assert_eq!(ayb::convert(input_digits, input_base, output_base).unwrap().as_slice(),
+               output_digits);
+}
+
+#[test]
+#[ignore]
+fn leading_zeros() {
+    let input_base = 7;
+    let input_digits = &[0, 6, 0];
+    let output_base = 10;
+    let output_digits = &[4, 2];
+    assert_eq!(ayb::convert(input_digits, input_base, output_base).unwrap().as_slice(),
+               output_digits);
+}
+
+#[test]
+#[ignore]
+fn negative_digit() {
+    let input_base = 2;
+    let input_digits = &[1, -1, 1, 0, 1, 0];
+    let output_base = 10;
+    assert!(ayb::convert(input_digits, input_base, output_base).is_err());
+}
+
+#[test]
+#[ignore]
+fn invalid_positive_digit() {
+    let input_base = 2;
+    let input_digits = &[1, 2, 1, 0, 1, 0];
+    let output_base = 10;
+    assert!(ayb::convert(input_digits, input_base, output_base).is_err());
+}
+
+#[test]
+#[ignore]
+fn first_base_is_one() {
+    let input_base = 1;
+    let input_digits = &[];
+    let output_base = 10;
+    assert!(ayb::convert(input_digits, input_base, output_base).is_err());
+}
+
+#[test]
+#[ignore]
+fn second_base_is_one() {
+    let input_base = 2;
+    let input_digits = &[1, 0, 1, 0, 1, 0];
+    let output_base = 1;
+    assert!(ayb::convert(input_digits, input_base, output_base).is_err());
+}
+
+#[test]
+#[ignore]
+fn first_base_is_zero() {
```

Comment:
I know that these test names come from the canonical set,  but I think "first" should be replaced with "input", and "second" should be replaced with "output", since that reflects the language used in the tests.